contracts/pol/CarbonPOL.sol

```diff
@@ -200,17 +198,19 @@ contract CarbonPOL is ICarbonPOL, Upgradeable, ReentrancyGuardUpgradeable, Utils
     /**
      * @inheritdoc ICarbonPOL
      */
-    function tokenPrice(Token token) public view returns (Price memory price) {
+    function tokenPrice(Token token) public view returns (Price memory) {
+        // cache trading start time to save gas
+        uint32 tradingStartTime = _tradingStartTimes[token];
         // return 0 if trading hasn't been enabled for a token
-        if (!_tradingEnabled(token)) {
+        if (tradingStartTime == 0) {
```

Comment:
Super nitpick (feel free to ignore and resolve): since this duplicated the logic of testing if trading is enabled (once here and once in `_tradingEnabled`), you may consider moving the logic to a shared pure function that receives the trading start time.